core/src/main/kotlin/fr/sncf/osrd/standalone_sim/SafetySpeed.kt

```diff
@@ -73,6 +95,24 @@ fun makeSafetySpeedRanges(
     return res.subMap(0.meters, chunkPath.length)
 }
 
+/** Check if a given stop is in a range of a given signaling system. */
+private fun isStopInSignalingSystemRange(
+    stopOffset: Offset<TravelledPath>,
+    signalingRanges: DistanceRangeMap<String>,
+    signalingSystem: String,
+): Boolean {
+    for (range in signalingRanges) {
+        if (
+            range.value == signalingSystem &&
+                range.lower <= stopOffset.distance &&
+                range.upper >= stopOffset.distance
+        ) {
+            return true
+        }
+    }
+    return false
```

Comment:
Actually, there is the `get()` on DistanceRangeMap:
```suggestion
    return (signalingRanges.get(stopOffset) == signalingSystem)
```

Reply:
Much better :)